### PR TITLE
Evaluate end-to-end performance

### PR DESCRIPTION
**Pull request by katerina-cropguard**
- a script to evaluate end-to-end performance
- updated Readme

### CODE REVIEW

**Review by simdage — APPROVED** [review]
LGTM

File: croptalk/evaluate_overall_performance.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+client = Client()
+results = client.run_on_dataset(
+    dataset_name=DATASET_NAME,
+    llm_or_chain_factory=MODEL,
+    evaluation=eval_config,
+    input_mapper=lambda x: {"question": x["Question"], 
+                            "chat_history": []},
+    # project_name="CropTalk AprDemo", # has to be unique
+    # tags=["simple_chain"],           # list of tags
```
Comment (dtourillon-cropguard) [inline croptalk/evaluate_overall_performance.py:34]: Out of curiosity, what are these commented lines? When should we uncomment them & what should we set them to?

File: croptalk/evaluate_overall_performance.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+load_dotenv('secrets/.env.secret')
+load_dotenv('secrets/.env.shared')
+
+DATASET_NAME = "sp-cp-bp-cih-70"
+
+# DOCS: 
+# https://api.python.langchain.com/en/latest/smith/langchain.smith.evaluation.config.RunEvalConfig.html
+
+eval_config = RunEvalConfig(
```
Comment (dtourillon-cropguard) [inline croptalk/evaluate_overall_performance.py:20]: An LLM will "decide" if ref answer and current answer are "close enough" to determine correctness, is that it? Do we know how "close enough" is defined exactly?

**Review by dtourillon-cropguard — APPROVED** [review]
2 questions, otherwise LGTM

File: croptalk/evaluate_overall_performance.py (hunk 1 above)
Comment (katerina-cropguard) [reply croptalk/evaluate_overall_performance.py:34]: By default, it's better to keep them commented. But if we want to create a unique interpretable experiment name, we can set those. It's just that the run will break if they are uncommented, but not attended to (at least the project name).

File: croptalk/evaluate_overall_performance.py (hunk 2 above)
Comment (katerina-cropguard) [reply croptalk/evaluate_overall_performance.py:20]: We have to read on that. For now, I'm using default QA evaluator from langsmith. We definitely can improve it and create custom evaluators later on

**Pull request merged**